src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftExec.java

```diff
@@ -59,6 +61,39 @@ default String getDisplayName() {
     default long getGlobalTimeoutConfiguration() {
         return GlobalConfig.getExecWait();
     }
+    
+    default IClient getClient(TaskListener listener, String displayName, Map<String, String> overrides, String token) {
+        Auth auth = getAuth();
+        ClientBuilder cb = null;
+        long timeoutInMS = this.getTimeout(listener, Boolean.parseBoolean(getVerbose(overrides)), overrides);
+        if (timeoutInMS > Integer.MAX_VALUE) {
+            int timeoutInMin = (int) (timeoutInMS / 60000);
+            cb = new ClientBuilder(getApiURL(overrides)).
+                    sslCertificateCallback(auth).
+                    withReadTimeout(timeoutInMin, TimeUnit.MINUTES).
+                    withWriteTimeout(timeoutInMin, TimeUnit.MINUTES).
+                    withConnectTimeout(timeoutInMin, TimeUnit.MINUTES).
+                    usingToken(token).
+                    sslCertificate(getApiURL(overrides), auth.getCert());
+        } else {
+            cb = new ClientBuilder(getApiURL(overrides)).
+                    sslCertificateCallback(auth).
+                    withReadTimeout((int)timeoutInMS, TimeUnit.MILLISECONDS).
+                    withWriteTimeout((int)timeoutInMS, TimeUnit.MILLISECONDS).
+                    withConnectTimeout((int)timeoutInMS, TimeUnit.MILLISECONDS).
+                    usingToken(token).
+                    sslCertificate(getApiURL(overrides), auth.getCert());
+        }
+        
+        if (auth.useCert())
+            cb.sslCertCallbackWithDefaultHostnameVerifier(true);
```

Comment:
mostly lgtm, but what was the default behavior before this change?  especially around the hostname verification, since you seem to be introducing default behavior for that here now.